opensrp-chw-malaria/src/main/java/org/smartregister/chw/malaria/provider/MalariaRegisterProvider.java

```diff
@@ -64,42 +71,70 @@ public void getView(Cursor cursor, SmartRegisterClient smartRegisterClient, Regi
-                viewHolder.patientColumn.performClick();
-            }
-        });
+        try {
+            String ageString;
+            String fname = getName(
+                    Utils.getValue(pc.getColumnmaps(), DBConstants.KEY.FIRST_NAME, true),
+                    Utils.getValue(pc.getColumnmaps(), DBConstants.KEY.MIDDLE_NAME, true));
 
-        viewHolder.registerColumns.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                viewHolder.dueButton.performClick();
+            String dobString = Utils.getValue(pc.getColumnmaps(), DBConstants.KEY.DOB, false);
+            int age = new Period(new DateTime(dobString), new DateTime()).getYears();
+
+            if (age == 0) {
+                ageString = Utils.getDuration(Utils.getValue(pc.getColumnmaps(), DBConstants.KEY.DOB, false));
+            } else {
+                ageString = age + "";
             }
-        });
 
+            String patientName = getName(fname, Utils.getValue(pc.getColumnmaps(), DBConstants.KEY.LAST_NAME, true));
+            viewHolder.patientName.setText(String.format("%s, %s", patientName, ageString));
+            viewHolder.textViewGender.setText(Utils.getValue(pc.getColumnmaps(), DBConstants.KEY.GENDER, true));
+            viewHolder.textViewVillage.setText(Utils.getValue(pc.getColumnmaps(), DBConstants.KEY.VILLAGE_TOWN, true));
+
+            viewHolder.patientColumn.setOnClickListener(onClickListener);
+            viewHolder.patientColumn.setTag(pc);
+            viewHolder.patientColumn.setTag(R.id.VIEW_ID, BaseMalariaRegisterFragment.CLICK_VIEW_NORMAL);
+
+            Date date = new SimpleDateFormat("dd-MM-yyyy", Locale.getDefault()).parse(Utils.getValue(pc.getColumnmaps(), DBConstants.KEY.MALARIA_TEST_DATE, false));
+            Days days = Days.daysBetween(new LocalDateTime(date), LocalDateTime.now());
+            dueColorSequence(days.getDays(), viewHolder);
+
+            viewHolder.dueButton.setOnClickListener(onClickListener);
+            viewHolder.dueButton.setTag(pc);
+            viewHolder.dueButton.setTag(R.id.VIEW_ID, BaseMalariaRegisterFragment.CLICK_VIEW_NORMAL);
+            viewHolder.registerColumns.setOnClickListener(onClickListener);
+
+            viewHolder.registerColumns.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    viewHolder.patientColumn.performClick();
+                }
+            });
+
+            viewHolder.registerColumns.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    viewHolder.dueButton.performClick();
+                }
+            });
+
+
+        } catch (Exception e) {
+            Timber.e(e);
+        }
+    }
+
+    private void dueColorSequence(int testDate, RegisterViewHolder viewHolder) {
+        if (testDate < 7) {
```

Comment:
if testDate was say 6 would all the conditions not evaluate to true?

Reply:
No they wont, because of the way if conditions work.

Reply:
cool